Physical_Engine/Query/BottomCentreline.cs

```diff
@@ -0,0 +1,65 @@
+﻿using BH.Engine.Geometry;
+using BH.Engine.Reflection;
+using BH.Engine.Spatial;
+using BH.oM.Base;
+using BH.oM.Geometry;
+using BH.oM.Physical.FramingProperties;
+using BH.oM.Reflection.Attributes;
+using System;
+using System.Collections.Generic;
+using System.ComponentModel;
+using System.Configuration;
+using System.Linq;
+using System.Linq.Expressions;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace BH.Engine.Physical
+{
+    public static partial class Query
+    {
+
+        [Description("Returns the bottom centreline of an IFramingElement")]
+        [Input("BHoM.Physical.Elements.IFramingElement", "The IFramingEment to query the bottom centreline of")]
+        [Output("ICurve", "The bottom centreline of the IFramingElement")]
+
+        public static ICurve BottomCentreline(this BH.oM.Physical.Elements.IFramingElement element)
+        {
+            ICurve location = element.Location;
+
+            Vector normal;
+            
+            if(location.IIsLinear())
+            { 
+                normal = BH.Engine.Physical.Query.Normal(element);
+            }
+            else
+            {
+                Engine.Reflection.Compute.RecordError("IFramingElement must have linear location line");
+                normal = null;
+            }
+
+            double height = 0;
+           
+            object propValue = element.PropertyValue("Property.Profile.Height");
```

Comment:
mostly as an FYI here. Think this case actually can be ok for extracting height like this, using the PropertyValue method, but should be used with quite a bit of care as it is significantly slower than straight access to the properties. Again, think this might be one of the corner cases that it is ok to use it, given how many levels of abstraction you have to go through, but FYI. 😄

Reply:
The level of abstraction is only 3 layers, I thought all `IFramingElements` had a property called `Property` which would then make this easier to just do `double height = element.Property.Profile.Height`?

Reply:
@FraserGreenroyd I think since different types of IFramingElements have different properties writing element.Property.Profile.Height will not work, I did not get it to work anyways.

Reply:
Just double checked the code, it seems all `IFramingElement` have an `IFramingElementProperty` but that interface does not enforce a `IProfile` property. However, we only have one object which implements `IFramingElementProperty` which is `ConstantFramingElement`. @IsakNaslundBh should `IFramingElementProperty` enforce an `IProfile` property?

If not then, this approach is fine, but I would probably go for a quicker convert than the `if`/`else` you've done instead.

Reply:
>  @IsakNaslundBh should IFramingElementProperty enforce an IProfile property?

No, can not. need to be more flexible.

Also, all profiles do not have a `Height` property. Look at tube and circle profiles for example.

Discussed offline with @annastigenberg that a solution around it would be to add `FullHeight` query methods for both `IFramingProperty` as well as `IProfile`. Something that I am not necessarily against, but for sure expands the scope of this task.

Another way to do it, that just came to mind, is to check if the property is a `ConstantFramingProeprty`, if so, extract the profile, then extract the edges of the profile, then check the bounds of them, and then you can get the height from the difference in Y coordinates of the min and the max.

Might actually be a better interim solution than relying on the reflection in PropertyValue

Reply:
> Another way to do it, that just came to mind, is to check if the property is a ConstantFramingProeprty, if so, extract the profile, then extract the edges of the profile, then check the bounds of them, and then you can get the height from the difference in Y coordinates of the min and the max.

I think this is a good fix for this PR, then agree a new issue raised and further work on `FullHeight` queries - which I agree is out of scope for this PR.

Reply:
The definition of the IFramingElement has an ICurve and an IFramingElementProperty, which in its turn does not have ConstantFramingProperty. Therefore not sure how to or if it is possible to access the edges of the profile.

Reply:
The IFramingElementProperty can _be_ a ConstantFramingProperty.

What you can check is something like

```
If(element.Property is ConstantFramingProperty)
{
     ConstantFramingProperty constantProperty = element.Property as ConstantFramingProperty;
IProfile profile = constantProperty.Profile;

......
}
else
{
//Raise warning stating this is only wokring for constatn proeprties
}
```
Sure I have some typos in the above, but that is the principles of it.